src/sknnr/datasets/_base.py

```diff
@@ -4,7 +4,7 @@
 import sys
 from dataclasses import dataclass
 from importlib import resources
-from typing import TYPE_CHECKING, TextIO, Union
+from typing import TYPE_CHECKING, TextIO
```

Comment:
Generic comment/question (not just this line).  I assume we'll need to use `from __future__ import annotations` on each module that we want to use the newer typing syntax, correct?  My brain wants to have this statement in one module and then import that module where we use it, but I don't think that saves us any effort and may actually add to our load.

Reply:
> I assume we'll need to use from __future__ import annotations on each module that we want to use the newer typing syntax, correct?

Yes. For typing with built-in types (e.g. `list`) we can do that automatically once we go to [3.9](https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections), but using `|` instead of `Union` won't happen until [3.10](https://peps.python.org/pep-0604/), and as we figured out before, forward references are still indefinite. I'm guessing you're right and we'll end up needing the `__future__` import in just about every module for the foreseeable future.

I also had the idea of trying to automatically include it in all our modules somehow (I think I dislike the import because it's never explicitly used), but I landed at the same spot as you. I don't think there's any good way to do that, and even if there was I think it would be too surprising to silently modify behavior.